src/librustc_mir/build/matches/mod.rs

```diff
@@ -66,125 +137,89 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
             .collect();
 
-        let mut has_guard = false;
-
-        // assemble a list of candidates: there is one candidate per
-        // pattern, which means there may be more than one candidate
-        // *per arm*. These candidates are kept sorted such that the
-        // highest priority candidate comes first in the list.
-        // (i.e., same order as in source)
-
-        let candidates: Vec<_> = arms.iter()
-            .enumerate()
-            .flat_map(|(arm_index, arm)| {
-                arm.patterns
-                    .iter()
-                    .enumerate()
-                    .map(move |(pat_index, pat)| (arm_index, pat_index, pat, arm.guard.clone()))
-            })
-            .zip(
-                pre_binding_blocks
-                    .iter()
-                    .zip(pre_binding_blocks.iter().skip(1)),
-            )
-            .map(
-                |(
-                    (arm_index, pat_index, pattern, guard),
-                    (pre_binding_block, next_candidate_pre_binding_block)
-                )| {
-                    has_guard |= guard.is_some();
-
-                    // One might ask: why not build up the match pair such that it
-                    // matches via `borrowed_input_temp.deref()` instead of
-                    // using the `discriminant_place` directly, as it is doing here?
-                    //
-                    // The basic answer is that if you do that, then you end up with
-                    // accceses to a shared borrow of the input and that conflicts with
-                    // any arms that look like e.g.
-                    //
-                    // match Some(&4) {
-                    //     ref mut foo => {
-                    //         ... /* mutate `foo` in arm body */ ...
-                    //     }
-                    // }
-                    //
-                    // (Perhaps we could further revise the MIR
-                    //  construction here so that it only does a
-                    //  shared borrow at the outset and delays doing
-                    //  the mutable borrow until after the pattern is
-                    //  matched *and* the guard (if any) for the arm
-                    //  has been run.)
-
-                    Candidate {
-                        span: pattern.span,
-                        match_pairs: vec![MatchPair::new(discriminant_place.clone(), pattern)],
-                        bindings: vec![],
-                        ascriptions: vec![],
-                        guard,
-                        arm_index,
-                        pat_index,
-                        pre_binding_block: *pre_binding_block,
-                        next_candidate_pre_binding_block: *next_candidate_pre_binding_block,
-                    }
-                },
-            )
-            .collect();
-
+        // There's one move pre_binding block that there are candidates so that
```

Comment:
I don't know what this sentence means. Specifically, I don't understand the words "that there are candidates" means in the context of the sentence.